### PR TITLE
adds slapping!

### PR DESCRIPTION
**Pull request by anconfuzedrock**
:cl: rock
soundadd: you can now harmlessly slap somebody by aiming for the mouth on disarm intent.
soundadd: you can only slap somebody who is unarmed on help intent, restrained, or ready to slap you.
/:cl:
Shouldn't cause any issues because the target must be on help intent AND not holding something active handedly, or they can be restrained (aggressive grabs count) or preparing for a slap fight by targetting the mouth themselves.

### CODE REVIEW

**Comment by kevinz000** [comment]
>making disarms fail when targetting head as people wil lhave help intent most of the time
no.

**Comment by anconfuzedrock** [comment]
who the fuck targets the head? I'd certainly make it more specific so the target must be unarmed if I could figure out how.

**Comment by anconfuzedrock** [comment]
sorry, I shouldn't have offered to improve it.

**Comment by Kromgar** [comment]
b-baka

**Comment by PKPenguin321** [comment]
no more of these

**Comment by ExcessiveUseOfCobblestone** [comment]
To keep the meme PR going this should unironically deconvert revs ['slap' out of it!]

**Comment by anconfuzedrock** [comment]
that actually sounds ok

**Comment by dannno** [comment]
>who the fuck targets the head?

me?? probably a relatively significant amount of people??

**Comment by anconfuzedrock** [comment]
why though its useless now it can't knock out I'll try to make it more specific now here I go

**Comment by dannno** [comment]
It's not useless, it's specifically targeting a part of the body, it's as useful as targeting anywhere else
losing that for something like this is stupid

**Comment by anconfuzedrock** [comment]
exclusively for disarming unarmed to the head though, I will make it more specific though.

File: code/modules/mob/living/carbon/human/species.dm [hunk 1]
```diff
@@ -994,7 +994,14 @@
 
 
 /datum/species/proc/disarm(mob/living/carbon/human/user, mob/living/carbon/human/target, datum/martial_art/attacker_style)
-	if(target.check_block())
+	if(check_zone(user.zone_selected) == "head" && (target.a_intent == INTENT_HELP))
+		playsound(target.loc, 'sound/weapons/slap.ogg', 50, 1, -1)
+		user.visible_message("<span class='danger'>[user] slaps [target] in the face!</span>",
+			"<span class='notice'> You slap [target] in the face! </span>",\
+		"You hear a slap.")
+		target.endTailWag()
```
Comment (ShadowDeath6) [inline code/modules/mob/living/carbon/human/species.dm]: I know why this was added, now

**Comment by ShadowDeath6** [comment]
bad idea to make a harmful intent do literally nothing and fail if you target a commonly targeted area of the body

**Comment by kilkun** [comment]
this is cool and good.

anconfuzedrock,  a good compromise make it function like a disarm but it just has a special flavor text/sound/maybe also smacks smokeables out of people's mouths.

**Comment by anconfuzedrock** [comment]
you made me think nobody targets the mouth

**Comment by Kromgar** [comment]
Mouth works. The only time you target mouth is point blank fire

**Comment by Shapsy** [comment]
If you don't think slapping people upside the head wouldn't be used to great comedic effect, get outta here

**Comment by kilkun** [comment]
>the only time you target mouth is point blank fire
or if you want to have a better chance at punch stuns
or if you want to bypass 90% of helmets iirc

**Comment by ExcessiveUseOfCobblestone** [comment]
stop telling people secrets spaceman this is goon now

**Comment by Shapsy** [comment]
>or if you want to have a better chance at punch stuns

pretty sure that's not actually a thing

**Comment by optimumtact** [comment]
none of things he mentioned work or should work

**Comment by ChemicalRascal** [comment]
@optimumtact sounds like it's time for a new feature!

**Comment by ShadowDeath6** [comment]
Slippery slope is too steep, it's been less than a day. 

Preference to slap someone automatically when they try to pat you when?

**Comment by bgobandit** [comment]
why does the lizard keep coding potential fetish shit 

makes you think

**Comment by JJRcop** [comment]
Hey depending on the active hand this should change the target's dir

**Comment by RandomMarine** [comment]
Imagine assistants getting into slapfights.

That should be reason enough to be merged.

**Comment by anconfuzedrock** [comment]
this is like the opposite of fetish shit what the fuck slap IN THE FACE

**Comment by optimumtact** [comment]
bandit was probably slapped a lot as a child

**Comment by Grazyn** [comment]
>grabs cute medbay stutter whore's ass
>she can't slap me back because I'm on grab intent

meh

**Comment by anconfuzedrock** [comment]
Good point, I spent 2 hours trying to add another way to slap somebody with another intent and failed, i think i might know why though and will figure it out later

**Comment by PKPenguin321** [comment]
Just make this an emote

File: code/modules/mob/living/carbon/human/species.dm (hunk 1 above)
Comment (Supermichael777) [inline code/modules/mob/living/carbon/human/species.dm]: 🤕

**Comment by ExcessiveUseOfCobblestone** [comment]
wait this ignores block check lol

**Comment by bgobandit** [comment]
>this is like the opposite of fetish shit what the fuck slap IN THE FACE

have you not seen this thing called "most pornography"

**Comment by anconfuzedrock** [comment]
How would i deal with the block check

**Comment by anconfuzedrock** [comment]
that was an accident

**Comment by anconfuzedrock** [comment]
apparently it is set to return "mouth" as "head" meaning 2 things- first, targetting mouth IS the same thing as targetting head for combat, and second, I need to fix this

**Comment by anconfuzedrock** [comment]
PSA: do not use check_zone. it will ruin everything.

**Comment by anconfuzedrock** [comment]
alright time you would want to disarm somebody in the scenario possible now is if the murderboner is holding their ebow in their off hand AND they are on help intent AND you are aiming for the mouth

**Comment by anconfuzedrock** [comment]
alright the only possible flaw I can see now is complicated situations in which you would try to slap but be unable to.

**Comment by anconfuzedrock** [comment]
yea this is definitely ready

**Comment by PKPenguin321** [comment]
should really be an emote

**Comment by optimumtact** [comment]
It can't be an emote because emotes have no way of targeting someone to carry out the slap and appropriate interactions.

**Comment by PKPenguin321** [comment]
sure it can just check for carbon/human in range 1 or whatever
if there's more than one you can just slap them all, do a windmill slap

**Comment by optimumtact** [comment]
wow great idea, lets make you headcoder so you can drive this truck off a bridge next

**Comment by ShadowDeath6** [comment]
wow, looks like this orange is a bit sour, maybe they're not in season yet. ;^)

**Comment by RemieRichards** [comment]
>It can't be an emote because emotes have no way of targeting someone to carry out the slap and appropriate interactions.

They do though? a lot of emotes accept a name after them, after which they look for a mob nearby with that name.

**Comment by Grazyn** [comment]
>They do though? a lot of emotes accept a name after them, after which they look for a mob nearby with that name.

And if you don't specify a target you just slap yourself.

**Comment by anconfuzedrock** [comment]
Maybe i shouldve done an emote in the firet place but I didnt know it was possible to givr it special effexts like the slap sound and >emotes 
Not like it will cause any accidents in its current state, anyway.

**Comment by RemieRichards** [comment]
just make it an emote

**Comment by anconfuzedrock** [comment]
I don' see the point at this point, itd if anything make it harder to use and with the requirements I put in it won't interfere with anything, not worth the effort, plus an emote would be spammable.

**Comment by RemieRichards** [comment]
Something like this is PERFECT for an emote, it's absolutely atrocious as a random override in the middle of a disarm.

**Comment by PKPenguin321** [comment]
I could look at refactoring it later today maybe

**Comment by Hatfish** [comment]
You can't have a good slapfight with emotes though 
Unless you're a filthy RPer, that is.

**Comment by ShadowDeath6** [comment]
You can't have a good slapfight with this, either. Someone has to be on help intent.

**Comment by anconfuzedrock** [comment]
I wouldn't have a problem with it being an emote if somebody could give me a way to make it auto slap somebody if they are next to you if you don't specify instead of yourself and a way to make it not spammable

**Comment by anconfuzedrock** [comment]
yea I'd certainly give it the mindswap/old ei nath style menu if I could but as it stands that seems more than difficult, if anyone can give me a simple way to do that I'd do it but I'll leave it as is otherwise.

**Comment by optimumtact** [comment]
If this should be an emote then so should patting

**Comment by PKPenguin321** [comment]
Yeah it should be

**Comment by anconfuzedrock** [comment]
alright, now that slapfights are possible, the only real flaw I can see here is that this is a bit snowflakey. what oranges said though, and with that I think this is absolutely as ready as it can possibly be, I could not improve this any more.

**Comment by PKPenguin321** [comment]
You messed up your conflict fixing, you're supposed to leave every file ticked in the commit area

**Comment by anconfuzedrock** [comment]
what do you mean? I didn't have to fix any conflicts. where do you see a problem?

**Comment by anconfuzedrock** [comment]
the hell happened to this pr  I have no idea

**Comment by anconfuzedrock** [comment]
I have no idea what I just did but it fixed it, thanks oranges.

File: code/modules/mob/living/carbon/human/species.dm [hunk 2]
```diff
@@ -994,7 +994,14 @@
 
 
 /datum/species/proc/disarm(mob/living/carbon/human/user, mob/living/carbon/human/target, datum/martial_art/attacker_style)
-	if(target.check_block())
+	if((user.zone_selected) == "mouth" && (((target.a_intent == INTENT_HELP) && (!target.get_active_held_item())) || (target.restrained()) || (target.zone_selected) == "mouth"))
```
Comment (optimumtact) [inline code/modules/mob/living/carbon/human/species.dm]: Please break this into either one or two if checks or refactor it to be easier to read.

I'm not even sure you need the number of brackets you have

Comment (anconfuzedrock) [inline code/modules/mob/living/carbon/human/species.dm]: how would I do that I don't get what you are askin here

Comment (anconfuzedrock) [inline code/modules/mob/living/carbon/human/species.dm]: alright ill clean it up gimme an hour

Comment (optimumtact) [inline code/modules/mob/living/carbon/human/species.dm]: var/aim_for_mouth  = user.zone_selected == "mouth"
    var/target_on_help =  target.a_intent == INTENT_HELP
    var/restrained_or_not_holding = !target_get_active_held_item() || target.restrained()
    if(aim_for_mouth && target_on_help && restrained_or_not_holding)
       do your thing

**Comment by anconfuzedrock** [comment]
alright its great

**Pull request merged**